users/script.js

```diff
@@ -264,6 +272,107 @@ async function getParticularUserData(
   }
 }
 
+function addAvailibilityFilterOptions() {
+  const options = [
+    { value: 'none', text: NONE },
+    { value: 'active', text: ACTIVE },
+    { value: 'ooo', text: OOO },
+    { value: 'idle', text: IDLE },
+  ];
+
+  const fragment = document.createDocumentFragment();
+
+  options.forEach((option) => {
+    const optionElement = document.createElement('option');
+    optionElement.value = option.value;
+    optionElement.text = option.text;
+    fragment.appendChild(optionElement);
+  });
+
+  availabilityFilter.appendChild(fragment);
+  availabilityFilter.options[0].disabled = true;
+  availabilityFilter.options[0].selected = true;
+}
+
+async function getUsersStatusData(state) {
+  try {
+    const usersRequest = await makeApiCall(
+      `${RDS_API_USERS}status?state=${state}`,
+    );
+    const currentstate = await usersRequest.json();
+    return currentstate.allUserStatus;
+  } catch (err) {
+    throw new Error(`User list request failed with error: ${err}`);
+  }
+}
+
+function showUserList(users) {
+  const ulElement = document.createElement('ul');
+
+  users.forEach((userData) => {
+    const listElement = document.createElement('li');
+    const imgElement = document.createElement('img');
+    imgElement.src = userData.picture?.url ?? DEFAULT_AVATAR;
+    imgElement.classList.add('user-img-dimension');
+    const pElement = document.createElement('p');
+    const node = document.createTextNode(`${userData.full_name}`);
+    pElement.appendChild(node);
+    listElement.appendChild(imgElement);
+    listElement.appendChild(pElement);
+
+    if (tileViewActive) {
+      let imgElement = listElement.firstChild;
+      listElement.classList.remove('tile-width');
+      imgElement.classList.add('remove-element');
+    }
+    listElement.onclick = () => {
+      document.getElementById('user-search').value = '';
+      window.location.href = `details/index.html?username=${userData.username}`;
+    };
+    ulElement.appendChild(listElement);
+    paginationElement.classList.add('remove-element');
+    paginationElement.classList.remove('pagination');
+  });
+
+  userListElement.innerHTML = '';
+  userListElement.appendChild(ulElement);
+}
+
+async function filterUserByAvailability() {
+  const availabilityFilterValue = availabilityFilter.value;
+  if (availabilityFilterValue === 'idle') {
```

Comment:
`Suggestion [for-next-time]:` For better readibility if number of if conditions are less.

```js
if(idle)
{
//your code here
return await someMethod();
}

if(active)
{
//your code here
return await someMethod();
}

//your code here
return await someMethod();

Reply:
changed